src/NHapi.Base/Parser/StructureDefinition.cs

```diff
@@ -7,17 +7,26 @@
     /// <inheritdoc />
     public class StructureDefinition : IStructureDefinition
     {
-        private HashSet<string> allChildrenNames;
+        // TODO: Replace locks with Lazy{T} when support for net35 is dropped.
+        private readonly object allChildrenNamesLock = new object();
 
-        private HashSet<string> allFirstLeafNames;
+        // TODO: Replace locks with Lazy{T} when support for net35 is dropped.
+        private readonly object allPossibleFollowingLeavesLock = new object();
+
+        // TODO: Replace locks with Lazy{T} when support for net35 is dropped.
```

Comment:
Idea: Create an issue that captures dropping support for net35, and then reference that issue here with the `TODO`.

Reply:
Makes sense, will do this.

Reply:
Created issue #253